Campus Density/DiningCells/DetailControllerHeaderCell.swift

```diff
@@ -0,0 +1,125 @@
+    var timeString: NSMutableAttributedString!
+    var timeLabel: UILabel!
+    var bottomBorder: UIView!
+
+    let spacing = 5
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupViews()
+    }
+
+    func setupViews() {
+
+        timeLabel = UILabel()
+        addSubview(timeLabel)
+
+        displayLabel = UILabel()
+        displayLabel.textColor = .black
+        displayLabel.textAlignment = .center
+        displayLabel.font = .thirtyBold
+        addSubview(displayLabel)
+
+        bottomBorder = UIView()
+        bottomBorder.backgroundColor = .densityDarkGray
+        addSubview(bottomBorder)
+
+        setupConstraints()
+    }
+
+    func setupConstraints() {
+        displayLabel.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(-5)
+            make.centerX.equalToSuperview()
+        }
+
+        timeLabel.snp.makeConstraints { make in
+            make.top.equalTo(displayLabel.snp.bottom).offset(spacing)
+            make.centerX.equalToSuperview()
+        }
+
+        bottomBorder.snp.makeConstraints { make in
+            make.bottom.equalToSuperview()
+            make.height.equalTo(1)
+            make.width.equalToSuperview()
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(with place: Place) {
+        self.place = place
+        displayLabel.text = place.displayName
+        if place.displayName.count >= 23 {
+            displayLabel.font = .twentyEightBold
+        }
+
+        if place.isClosed {
+            timeString = NSMutableAttributedString(string: "Closed")
+            timeString.addAttribute(.foregroundColor, value: UIColor.orangeyRed, range: NSRange(location: 0, length: timeString.mutableString.length))
+        } else {
+            let closingString = getNextClosingText()
+            timeString = NSMutableAttributedString(string: "Open until `\(closingString)")
+            timeString.addAttribute(.foregroundColor, value: UIColor.lightTeal, range: NSRange(location: 0, length: timeString.mutableString.length))
+        }
+
+        timeLabel.attributedText = timeString
+    }
+
+    /**
+     Returns a string representing the next closing time for this place, with the hh:mm aa format.
+    */
+    func getNextClosingText() -> String {
+
+        let ithacaTime = TimeZone(identifier: "America/New_York")!
+        var ithacaCalendar = Calendar.current
+
+        ithacaCalendar.timeZone = ithacaTime
+
+        let currTime = Date()
+        let day = ithacaCalendar.component(.weekday, from: currTime)
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "hh:mm aa"
+        dateFormatter.timeZone = ithacaTime
+        guard let placeHoursStringified = place.hours[day - 1] else { return "Operating hours could not be found" }
+        let placeHours = placeHoursStringified.split(separator: "\n")
```

Comment:
At some point we should consider changing `API` so the hours are saved as HOURS and not as one huge processed string so you don't have to do this kind of getting time through string manipulation. Also separate concerns so only the hours cell worries about the final representation to the user... yuck

Reply:
I completely agree with this and was going to do write this piece of code in the hours API call itself. But I decided against it intentionally simply because that refactoring will not be having any long-term benefits. Instead, we'd be changing that API call and everywhere else it's used (like dining detail). In the latter case you'd end up writing code to parse dates => strings. There's processing either way. Also, this is the only time we'll be using this API call since gyms has a completely different endpoint for handling that where we can keep this in mind. What do you think?

Reply:
Yeah, I guess if it ain't broke don't fix it. We can just leave it as is for now and if we ever have the time we can change it but it's low priority compared to the other things